pkg/roachpb/span_config.proto

```diff
@@ -28,6 +26,23 @@ message GCPolicy {
   // before garbage collection. A value <= 0 means older versions are never
   // GC-ed.
   int32 ttl_seconds = 1 [(gogoproto.customname) = "TTLSeconds"];
+
+  // ProtectionPolicies is a list of policies that dictate GC behavior for a
+  // range (in conjunction with the GC TTL). A ProtectionPolicy can be used
+  // to indicate a timestamp above which GC should not run, regardless of the
+  // GC TTL.
```

Comment:
[nit] Maybe another sentence/phrase about how the data the config applies to is not GC-ed provided it wasn't already GC-ed pre config-application?

Reply:
Done.